src/read/macho/section.rs

```diff
@@ -102,6 +102,44 @@ where
             .data(self.file.endian, self.internal.data)
             .read_error("Invalid Mach-O section size or offset")
     }
+
+    // Try Go-style "ZLIB" header decompression. This is similar to the GNU-style
+    // format, but the header uses a slightly different format with a 4-byte "ZLIB"
+    // magic followed by a 64-bit big-endian uncompressed size.
```

Comment:
The format looks identical to the GNU-style: GNU use a 4 byte ZLIB magic followed by a 64 bit size, but in the ELF parser we have chosen to limit the size to 32 bits, which we do by checking that the upper 4 bytes of the size are zero.

It'd be nice for both of these formats to handle this the same way and share this code.

Reply:
Oh! Now I feel silly, I saw elsewhere the GNU-style thing described as 4 bytes of padding followed by a big-endian u32 ... which literally is a big-endian u64 that fits in u32. My bad.

I've extracted the logic from the elf module into a shared module and used it in both places. A downside I couldn't figure out how to fix without doing something drastic was that I had to strip the section type information from the error messages to reduce the duplication.

If you'd prefer I just keep two copies of the logic around, that's fine. Happy to go whatever route you think is best.

Reply:
The error message changes are fine.